crepecake/src/main/java/com/google/cloud/tools/crepecake/image/ReferenceLayer.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.blob.BlobDescriptor;
+
+/**
+ * A {@link Layer} reference that <b>does not</b> have the underlying content. It references the
+ * layer with its digest, size, and diff ID.
+ */
+public class ReferenceLayer extends Layer {
+
+  /** The {@link BlobDescriptor} of the compressed layer content. */
+  private final BlobDescriptor blobDescriptor;
+
+  /** The digest of the uncompressed layer content. */
+  private final DescriptorDigest diffId;
+
+  /** Instantiate with a {@link BlobDescriptor} and diff ID. */
+  public ReferenceLayer(BlobDescriptor blobDescriptor, DescriptorDigest diffId) {
```

Comment:
Why is the DescriptorDigest independent of the BlobDescriptor instead of part of it? Could a single Blob have different descriptors?

Reply:
A layer BLOB has its main DescriptorDigest that is a hash of the compressed layer tarball archive, but it also has a DescriptorDigest diff ID that is a hash of the uncompressed layer tarball archive.